### PR TITLE
Adding task setters, addresses #232

### PR DESCRIPTION
**Pull request by tarrencev**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @tarrencev has signed the CLA. Thanks for the pull request!

File: src/main/java/com/box/sdk/BoxTask.java [hunk 1]
```diff
@@ -180,6 +180,15 @@ public Date getDueAt() {
         }
 
         /**
+         * Sets the task's message.
```
Comment (tarrencev) [inline src/main/java/com/box/sdk/BoxTask.java]: Update

File: src/main/java/com/box/sdk/BoxTask.java [hunk 2]
```diff
@@ -180,6 +180,15 @@ public Date getDueAt() {
         }
 
         /**
+         * Sets the task's due date.
+         * @param dueAt the task's due date.
+         */
+        public void setDueAt(Date dueAt) {
+            this.dueAt = dueAt;
+            this.addPendingChange("due_at", dueAt.toString());
```
Comment (gcurtis) [inline src/main/java/com/box/sdk/BoxTask.java]: Use `BoxDateFormat` to ensure that the date is formatted in a way that the API can parse.

**Comment by gcurtis** [comment]
Looks good, just one comment about date formatting.

File: src/main/java/com/box/sdk/BoxFile.java [hunk 3]
```diff
@@ -141,7 +141,7 @@ public BoxSharedLink createSharedLink(BoxSharedLink.Access access, Date unshareD
         }
 
         if (dueAt != null) {
-            requestJSON.add("due_at", dueAt.toString());
+            requestJSON.add("due_at", BoxDateFormat.format(dueAt));
```
Comment (tarrencev) [inline src/main/java/com/box/sdk/BoxFile.java]: Updated this to use `BoxDateFormat` too

**Comment by gcurtis** [comment]
Looks good!

**Pull request merged**